deepspeed/inference/engine.py

```diff
@@ -343,16 +336,36 @@ def load_model_with_checkpoint(self, r_module):
         def load(module, state_dict, prefix):
             args = (state_dict, prefix, {}, True, [], [], error_msgs)
             if hasattr(module, 'weight'):
+                if module.weight.data.is_meta:
+                    # meta tensor cannot be casted or copied to, so we need to replace it with a normal tensor here
+                    module.weight = torch.nn.parameter.Parameter(data=torch.empty_like(module.weight.data,
```

Comment:
Should this meta tensor replacement happen every time or only when using CPUAccelerator? In other words, does this PR change behavior for other accelerators?

Reply:
this replacement happens just once if the module weight is meta, after the replacement, it's not meta any more. and yes. it works for other accelerator because forward could not work if the weight is in meta device(no real data.) in every accelerators. I not only check in cpu but also in CUDA RTX8000

Reply:
This benefits all acclerators.  On CPU we need this change in order to run model with large weight on server with less host memory.   It could also benefit server with multiple CUDA device, reduce the host memory needed.   This work is also tracked seperately in the following PR.  The plan is merge #3102 first so this PR alone can be more CPU specific.
https://github.com/microsoft/DeepSpeed/pull/3102

Reply:
@sywangyi and @delock, thanks for the explanation. In that case, this sounds like a bug fix rather than a CPUAccelerator feature. Was this code working previously?

Reply:
yes, it's a bug fix. the previous checkpoint loading code in AutoTP path could not work with KI checkpoint json, and it's not sharded checkpoint loading just like KI.

Reply:
Awesome! Thanks for the confirmation and bug fix.